backend/src/database/image_controller.rs

```diff
@@ -1,35 +1,85 @@
 use crate::general_helpers::ENV_VARS;
+use uuid::Uuid;
+use sqlx::{postgres::{PgPoolOptions, PgQueryResult}, QueryBuilder, Pool, Postgres, Row};
 
-use super::types::image::Image;
+use super::{
+    types::{image::Image, post::Post},
+    helpers::{DATABASE_URL, ensure_tables_exist},
+};
 
 fn get_mock_data() -> Vec<Image> {
     return vec![
         Image {
             id: "1".to_string(),
-            timestamp: 0,
             data: vec![0, 1, 2],
+            timestamp: 0,
         },
         Image {
             id: "2".to_string(),
-            timestamp: 1,
             data: vec![0, 5, 10],
+            timestamp: 1,
         },
         Image {
             id: "3".to_string(),
-            timestamp: 2,
             data: vec![10, 20, 30],
+            timestamp: 2,
         },
     ];
 }
 
-pub fn read_images(ids: Vec<String>) -> Vec<Image> {
+pub async fn read_images(ids: Vec<String>) -> Result<Vec<Image>, sqlx::Error> {
     if ENV_VARS.use_mocked_database {
-        return get_mock_data()
+        return Ok(get_mock_data()
             .iter()
             .filter(|image| ids.contains(&image.id))
             .map(|image| image.clone())
-            .collect::<Vec<Image>>();
+            .collect::<Vec<Image>>());
     } else {
-        todo!("Implement this part of the database interaction");
+        return Ok(read_images_from_db(ids).await?);
     }
 }
+
+async fn read_images_from_db(ids: Vec<String>) -> Result<Vec<Image>, sqlx::Error> {
+    let pool = PgPoolOptions::new().max_connections(5).connect(DATABASE_URL).await?;
+
+    let mut querybuilder_stub = QueryBuilder::new("select * from images where ");
+    let finalized_querybuilder = querybuilder_stub.push(ids.join(" or "));
+    let sql_query = finalized_querybuilder.build();
+
+    let query_result = sql_query.fetch_all(&pool).await?;
+
+    Ok(query_result.iter().map(
+        // Issue: sqlx only supports up to 64 bit integers in the "get" method (else it complains
+        // that trait bound is not satisfied)
+        |row| Image {id: row.get::<String, _>("id"), data: row.get::<Vec<u8>, _>("data"), timestamp: row.get::<i64, _>("timestamp") as u128}
```

Comment:
Hvad sker der hvis typerne i tabellen ikke passer her

Reply:
der hvor vi typecaster i64 til u128 kan der i hvert fald opstå problemer. Dette program:
`
println!("{}", (-1 as i64) as u128);
`
Outputter 
`
340282366920938463463374607431768211455
`

Reply:
Kan vi ikke bruge u64 i stedet for i64